internal/backend/db/dbgorm/scheme/scheme.go

```diff
@@ -211,12 +223,26 @@ func ParseEventRecord(e EventRecord) (sdktypes.EventRecord, error) {
 
 type Env struct {
 	EnvID     string `gorm:"primaryKey"`
-	ProjectID string `gorm:"index"`
+	ProjectID string `gorm:"index;foreignKey"`
 	Name      string
+	DeletedAt gorm.DeletedAt
 
 	// {pid.uuid}/{name}. easier to detect dups.
 	// See OrgMember for more.
 	MembershipID string `gorm:"uniqueIndex"`
+
+	// just for the foreign key. Wihtout it gorm won't enforce it
+	Project Project
+}
+
+// enforce foreign keys constrains while soft-deleting
+func (e *Env) BeforeDelete(db *gorm.DB) error {
+	var count int64
+	db.Model(&Deployment{}).Where("deleted_at is NULL and env_id = ?", e.EnvID).Count(&count)
```

Comment:
make sure you have the relevant indexes on deployment

Reply:
added 2 indexes